src/object.ts

```diff
@@ -127,7 +152,7 @@ export default class ObjectSchema<
   TContext = AnyObject,
   TDefault = any,
   TFlags extends Flags = '',
-> extends Schema<MakeKeysOptional<TIn>, TContext, TDefault, TFlags> {
+> extends Schema<TIn, TContext, TDefault, TFlags> {
```

Comment:
is it safe to remove the optional key util here?

Reply:
Yes, we don't want to mislead what the `__outputType` actual result is going to be.  Its up to the constructor (or in our case, the `object` function) to force optional keys upon TIn where necessary.  This behavior was moved into the `TypeFromTarget` via the `MakePartial` type.